### PR TITLE
Add UseAssertIn lint rule to Fixit

### PR DESCRIPTION
**Pull request by dkteo**
## Summary
* Adding the Python lint rule for `assertIn/assertNotIn`.
* The reason we want to do this is to warn users to not make use of `assertTrue(x in y)/assertFalse(x in y)` which is deprecated in Py3.8. See [deprecated rules](https://docs.python.org/3.8/library/unittest.html#deprecated-aliases)

## Test Plan

Ran `tox -e py38 -- fixit.tests.UseAssertInRule`
```
GLOB sdist-make: /Users/dkteo/Fixit/setup.py
py37 inst-nodeps: /Users/dkteo/Fixit/.tox/.tmp/package/1/fixit-0.1.2.zip
py37 installed: alabaster==0.7.12,appdirs==1.4.4,appnope==0.1.0,argon2-cffi==20.1.0,async-generator==1.10,attrs==20.3.0,Babel==2.9.0,backcall==0.2.0,black==20.8b1,bleach==3.2.1,certifi==2020.11.8,cffi==1.14.4,chardet==3.0.4,click==7.1.2,codecov==2.1.10,coverage==5.3,decorator==4.4.2,defusedxml==0.6.0,diff-cover==4.0.1,distlib==0.3.1,docutils==0.16,entrypoints==0.3,filelock==3.0.12,fixit @ file:///Users/dkteo/Fixit/.tox/.tmp/package/1/fixit-0.1.2.zip,flake8==3.8.4,idna==2.10,imagesize==1.2.0,importlib-metadata==3.1.0,inflect==5.0.2,ipykernel==5.3.4,ipython==7.19.0,ipython-genutils==0.2.0,ipywidgets==7.5.1,isort==5.6.4,jedi==0.17.2,Jinja2==2.11.2,jinja2-pluralize==0.3.0,jsonschema==3.2.0,jupyter==1.0.0,jupyter-client==6.1.7,jupyter-console==6.2.0,jupyter-core==4.7.0,jupyterlab-pygments==0.1.2,libcst==0.3.14,MarkupSafe==1.1.1,mccabe==0.6.1,mistune==0.8.4,mypy-extensions==0.4.3,nbclient==0.5.1,nbconvert==6.0.7,nbformat==5.0.8,nbsphinx==0.8.0,nest-asyncio==1.4.3,notebook==6.1.5,packaging==20.7,pandocfilters==1.4.3,parso==0.7.1,pathspec==0.8.1,pexpect==4.8.0,pickleshare==0.7.5,pluggy==0.13.1,prometheus-client==0.9.0,prompt-toolkit==3.0.8,psutil==5.7.3,ptyprocess==0.6.0,py==1.9.0,pycodestyle==2.6.0,pycparser==2.20,pyflakes==2.2.0,Pygments==2.7.2,pyparsing==2.4.7,pyre-check==0.0.41,pyre-extensions==0.0.19,pyrsistent==0.17.3,python-dateutil==2.8.1,pytz==2020.4,pywatchman==1.4.1,PyYAML==5.3.1,pyzmq==20.0.0,qtconsole==5.0.1,QtPy==1.9.0,regex==2020.11.13,requests==2.25.0,Send2Trash==1.5.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx @ git+https://github.com/jimmylai/sphinx.git@971540b26a8412399a9face3752cafc4519ff748,sphinx-rtd-theme==0.5.0,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,terminado==0.9.1,testpath==0.4.4,toml==0.10.2,tornado==6.1,tox==3.20.1,traitlets==5.0.5,typed-ast==1.4.1,typing-extensions==3.7.4.3,typing-inspect==0.6.0,urllib3==1.26.2,virtualenv==20.2.1,wcwidth==0.2.5,webencodings==0.5.1,widgetsnbextension==3.5.1,zipp==3.4.0
py37 run-test-pre: PYTHONHASHSEED='2358965140'
py37 run-test: commands[0] | python -m unittest fixit.tests.UseAssertInRule
...............
----------------------------------------------------------------------
Ran 15 tests in 0.135s

OK
______________________________________________________________________________________ summary ______________________________________________________________________________________
  py37: commands succeeded
  congratulations :)
```

Ran `tox -e autofix`
```
GLOB sdist-make: /Users/dkteo/Fixit/setup.py
autofix inst-nodeps: /Users/dkteo/Fixit/.tox/.tmp/package/1/fixit-0.1.2.zip
autofix installed: alabaster==0.7.12,appdirs==1.4.4,appnope==0.1.0,argon2-cffi==20.1.0,async-generator==1.10,attrs==20.3.0,Babel==2.9.0,backcall==0.2.0,black==20.8b1,bleach==3.2.1,certifi==2020.11.8,cffi==1.14.4,chardet==3.0.4,click==7.1.2,codecov==2.1.10,coverage==5.3,decorator==4.4.2,defusedxml==0.6.0,diff-cover==4.0.1,distlib==0.3.1,docutils==0.16,entrypoints==0.3,filelock==3.0.12,fixit @ file:///Users/dkteo/Fixit/.tox/.tmp/package/1/fixit-0.1.2.zip,flake8==3.8.4,idna==2.10,imagesize==1.2.0,importlib-metadata==3.1.0,inflect==5.0.2,ipykernel==5.3.4,ipython==7.19.0,ipython-genutils==0.2.0,ipywidgets==7.5.1,isort==5.6.4,jedi==0.17.2,Jinja2==2.11.2,jinja2-pluralize==0.3.0,jsonschema==3.2.0,jupyter==1.0.0,jupyter-client==6.1.7,jupyter-console==6.2.0,jupyter-core==4.7.0,jupyterlab-pygments==0.1.2,libcst==0.3.14,MarkupSafe==1.1.1,mccabe==0.6.1,mistune==0.8.4,mypy-extensions==0.4.3,nbclient==0.5.1,nbconvert==6.0.7,nbformat==5.0.8,nbsphinx==0.8.0,nest-asyncio==1.4.3,notebook==6.1.5,packaging==20.7,pandocfilters==1.4.3,parso==0.7.1,pathspec==0.8.1,pexpect==4.8.0,pickleshare==0.7.5,pluggy==0.13.1,prometheus-client==0.9.0,prompt-toolkit==3.0.8,psutil==5.7.3,ptyprocess==0.6.0,py==1.9.0,pycodestyle==2.6.0,pycparser==2.20,pyflakes==2.2.0,Pygments==2.7.2,pyparsing==2.4.7,pyre-check==0.0.41,pyre-extensions==0.0.19,pyrsistent==0.17.3,python-dateutil==2.8.1,pytz==2020.4,pywatchman==1.4.1,PyYAML==5.3.1,pyzmq==20.0.0,qtconsole==5.0.1,QtPy==1.9.0,regex==2020.11.13,requests==2.25.0,Send2Trash==1.5.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx @ git+https://github.com/jimmylai/sphinx.git@971540b26a8412399a9face3752cafc4519ff748,sphinx-rtd-theme==0.5.0,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,terminado==0.9.1,testpath==0.4.4,toml==0.10.2,tornado==6.1,tox==3.20.1,traitlets==5.0.5,typed-ast==1.4.1,typing-extensions==3.7.4.3,typing-inspect==0.6.0,urllib3==1.26.2,virtualenv==20.2.1,wcwidth==0.2.5,webencodings==0.5.1,widgetsnbextension==3.5.1,zipp==3.4.0
autofix run-test-pre: PYTHONHASHSEED='4017664482'
autofix run-test: commands[0] | isort -q .
autofix run-test: commands[1] | black .
All done! ✨ 🍰 ✨
95 files left unchanged.
______________________________________________________________________________________ summary ______________________________________________________________________________________
  autofix: commands succeeded
  congratulations :)
```

### CODE REVIEW

**Comment by dkteo** [comment]
@jimmylai PTAL.
Builds seem to be failing due to lint errors:
```
ERROR: D:\a\Fixit\Fixit\fixit\rules\use_assert_in.py Imports are incorrectly sorted and/or formatted.
Skipped 2 files
ERROR: InvocationError for command 'D:\a\Fixit\Fixit\.tox\lint\Scripts\isort.EXE' --check-only . (exited with code 1)
```

But it works fine locally when I run isort manually.

**Comment by dkteo** [comment]
@mvismonte PTAL!

**Comment by jimmylai** [comment]
> @jimmylai PTAL.
> Builds seem to be failing due to lint errors:
> 
> ```
> ERROR: D:\a\Fixit\Fixit\fixit\rules\use_assert_in.py Imports are incorrectly sorted and/or formatted.
> Skipped 2 files
> ERROR: InvocationError for command 'D:\a\Fixit\Fixit\.tox\lint\Scripts\isort.EXE' --check-only . (exited with code 1)
> ```
> 
> But it works fine locally when I run isort manually.

You can fix the error by sorting the imports properly using command `tox -e autofix`.

**Comment by dkteo** [comment]
@jimmylai  thanks that did the trick.

Looks like the builds are still failing b/c of a test in the `UseClsInClassmethodRule`.
```
FAIL: test_INVALID_4 (fixit.common.testing.UseClsInClassmethodRule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/dkteo/Fixit/fixit/common/testing.py", line 223, in test_method
    return getattr(self, custom_test_method_name)(data, rule, fixture_file)
  File "/Users/dkteo/Fixit/fixit/common/testing.py", line 119, in _test_method
    validate_patch(report, test_case)
  File "/Users/dkteo/Fixit/fixit/common/testing.py", line 45, in validate_patch
    raise AssertionError(
AssertionError: Auto-fix did not produce expected result.
Expected:
class foo:
    # Do not replace in nested scopes.
    @classmethod
    async def cm(cls, b):
        b = cls
        b = lambda _: cls.__name__
        def g():
            return cls.__name__

        # Same-named vars in sub-scopes should not be replaced.
        **b = [a for a in [1,2,3]]**
        def f(a):
            return a + 1

But found:
class foo:
    # Do not replace in nested scopes.
    @classmethod
    async def cm(cls, b):
        b = cls
        b = lambda _: cls.__name__
        def g():
            return cls.__name__

        # Same-named vars in sub-scopes should not be replaced.
        **b = [cls for a in [1,2,3]]**
        def f(a):
            return a + 1

```
Seems like the auto replacement is replacing the first `a` in `b = [a for a in [1,2,3]]` to `cls` even though it shouldn't be.

**Comment by jimmylai** [comment]
> @jimmylai thanks that did the trick.
> 
> However, it looks like the builds are still failing b/c of a test in another rule: `UseClsInClassmethodRule`.
> 
> ```
> tox -e py38 -- fixit.tests.UseClsInClassmethodRule
> FAIL: test_INVALID_4 (fixit.common.testing.UseClsInClassmethodRule)
> ----------------------------------------------------------------------
> AssertionError: Auto-fix did not produce expected result.
> Expected:
> class foo:
>     # Do not replace in nested scopes.
>     @classmethod
>     async def cm(cls, b):
>         b = cls
>         b = lambda _: cls.__name__
>         def g():
>             return cls.__name__
> 
>         # Same-named vars in sub-scopes should not be replaced.
>         **b = [a for a in [1,2,3]]**
>         def f(a):
>             return a + 1
> 
> But found:
> class foo:
>     # Do not replace in nested scopes.
>     @classmethod
>     async def cm(cls, b):
>         b = cls
>         b = lambda _: cls.__name__
>         def g():
>             return cls.__name__
> 
>         # Same-named vars in sub-scopes should not be replaced.
>         **b = [cls for a in [1,2,3]]**
>         def f(a):
>             return a + 1
> ```
> 
> Seems like autofix is replacing the first `a` in `b = [a for a in [1,2,3]]` to `cls` even though it shouldn't be.

It's likely caused by a new change in new version of LibCST.
https://github.com/Instagram/LibCST/pull/413

There is a fix that @zsol is working on https://github.com/Instagram/LibCST/pull/423.

For now, to unblock this PR, we can pin LibCST version to the previous stable version `0.3.13`.
@dkteo you can pin the version in https://github.com/Instagram/Fixit/blob/master/requirements.txt
(`libcst==0.3.13`)

File: fixit/rules/use_assert_in.py
Comment (thatch) [inline fixit/rules/use_assert_in.py:16]: You can use reStructuredText here; please wrap to a reasonable line length too.

File: fixit/rules/use_assert_in.py
Comment (thatch) [inline fixit/rules/use_assert_in.py:20]: This isn't in the open-source version.

File: fixit/rules/use_assert_in.py
Comment (thatch) [inline fixit/rules/use_assert_in.py:69]: may as well find `assertFalse` too

File: fixit/rules/use_assert_in.py
Comment (dkteo) [reply fixit/rules/use_assert_in.py]: shortened the string length.

File: fixit/rules/use_assert_in.py
Comment (dkteo) [reply fixit/rules/use_assert_in.py]: thanks for catching. removed.

File: fixit/rules/use_assert_in.py
Comment (dkteo) [reply fixit/rules/use_assert_in.py:69]: makes sense.
this was taken from the internal rule though. i'll leave it for now so there's consistency between both places.

File: requirements.txt
Comment (jimmylai) [inline requirements.txt:2]: @Kronuz just released 0.3.15 today which should fix the issue.
@dkteo can you revert this change?

File: fixit/rules/use_assert_in.py
Comment (jimmylai) [reply fixit/rules/use_assert_in.py:72]: Yeah, use a single extract can be more effective and that can leave as future work. @dkteo can you add a `# TODO: ` comment to the code as a note?
Also, once this is merged, we'll want to remove the internal copy to avoid duplication since this will be released to internal dev environment.

**Review by jimmylai — APPROVED** [review]
Good job! Overall LGTM. Just some comments.

File: requirements.txt
Comment (dkteo) [reply requirements.txt:2]: will do!

**Comment by dkteo** [comment]
@jimmylai I've addressed the comments but am unable to merge b/c I don't have write access to the repo. Could you help me merge the PR? Thanks!

File: fixit/rules/use_assert_in.py
Comment (jimmylai) [inline fixit/rules/use_assert_in.py:11]: Simply import from fixit to be simpler. It's a convention we used.
```suggestion
from fixit import CstLintRule, InvalidTestCase as Invalid, ValidTestCase as Valid
```
@dkteo can you also fix this, then I can merge for you.

Comment (dkteo) [reply fixit/rules/use_assert_in.py]: done!

**Comment by jimmylai** [comment]
@dkteo, thanks for contributing!

**Pull request merged**
